src/mono/sample/wasi/native/Wasi.Native.Sample.csproj

```diff
@@ -8,6 +8,11 @@
     <WasmNativeStrip>false</WasmNativeStrip>
     <IsBrowserWasmProject>false</IsBrowserWasmProject>
     <WasmSingleFileBundle>true</WasmSingleFileBundle>
+    <OutputType>Library</OutputType>
+    <!-- This is needed for ReferenceCopyLocalPaths to include RuntimePackAsset-->
+    <CopyLocalLockFileAssemblies>true</CopyLocalLockFileAssemblies>
+    <!-- This is needed so it will not complain about WasmAssembliesToBundle is empty -->
+    <GenerateRuntimeConfigurationFiles>true</GenerateRuntimeConfigurationFiles>
```

Comment:
If the approach with SDK is working, we can move these defaults to https://github.com/dotnet/runtime/blob/main/src/mono/nuget/Microsoft.NET.Sdk.WebAssembly.Pack/build/Microsoft.NET.Sdk.WebAssembly.Pack.props#L4